ntp-proto/src/packet/mod.rs

```diff
@@ -270,9 +270,9 @@ impl<'a> NtpPacket<'a> {
         }
     }
 
-    pub fn deserialize(
+    pub fn deserialize<CP: CipherProvider>(
         data: &'a [u8],
-        cipher: Option<&Cipher>,
+        cipher: &CP,
```

Comment:
a matter of style, but you could use `cipher: &impl CipherProvider` here

Reply:
Done